include/gbinder_eventloop.h

```diff
@@ -0,0 +1,189 @@
+ * process (by definition).
+ *
+ * By default, GLib event loop is being used for callbacks and timeouts.
+ *
+ * It may be necessary to replace it with e.g. Qt event loop. Quite often
+ * Qt event loop is implemented by QEventDispatcherGlib which is sitting
+ * on top of GLib event and therefore works with the default implementation.
+ * But it won't work with e.g. QEventDispatcherUNIX.
+ *
+ * For Qt programs that use QEventDispatcherUNIX, it needs to be replaced
+ * with the one provided by libqbinder.
+ */
+typedef struct gbinder_eventloop_integration {
+
+    /**
+     * timeout_add
+     *
+     * Sets a function to be called at regular intervals (in milliseconds).
+     * If the function returns G_SOURCE_REMOVE, timeout is automatically
+     * destroyed (you must not call timeout_remove in this case). If the
+     * function returns G_SOURCE_CONTINUE, it will be called again after
+     * the same interval.
+     */
+    GBinderEventLoopTimeout* (*timeout_add)(guint millis, GSourceFunc func,
+        gpointer data);
+
+    /**
+     * timeout_remove
+     *
+     * Removes a pending timeout and destroys it. The caller makes sure that
+     * argument is not NULL. Note that timeout is automatically destroyed if
+     * the callback function returns G_SOURCE_REMOVE.
+     */
+    void (*timeout_remove)(GBinderEventLoopTimeout* timeout);
+
+    /**
+     * callback_new
+     *
+     * Creates a callback object. It returns you a reference, you must
+     * eventually pass the returned object to callback_unref to drop
+     * this reference.
+     *
+     * Note that it doesn't automatically schedule the callback. You
+     * must explicitly call callback_schedule to actually schedule it.
+     * The finalize function is invoked regardless of whether callback
+     * was cancelled or not.
+     */
+    GBinderEventLoopCallback* (*callback_new)(GBinderEventLoopCallbackFunc fun,
+        gpointer data, GDestroyNotify finalize);
+
+    /**
+     * callback_ref
+     *
+     * Increments the reference count. That prevents the object from being
+     * deleted before you drop this reference. The caller makes sure that
+     * argument is not NULL.
+     */
+    void (*callback_ref)(GBinderEventLoopCallback* cb);
+
+    /**
+     * callback_unref
+     *
+     * Decrements the reference count (drops the reference). When reference
+     * count reaches zero, the object gets deleted. The caller makes sure
+     * that argument is not NULL.
+     *
+     * Note that calling callback_schedule temporarily adds an internal
+     * reference until the callback is invoked or callback_cancel is called,
+     * whichever happens first.
+     */
+    void (*callback_unref)(GBinderEventLoopCallback* cb);
+
+    /**
+     * callback_schedule
+     *
+     * Schedules the callback to be invoked in the main loop at some point
+     * in the future (but as soon as possible). The caller makes sure that
```

Comment:
nit: The wording here does not mention interval => makes me think idle callbacks rather than timers.

Reply:
And it is idle callback rather than timer :)

Reply:
well, then, ok ;-)

but why not call it idle_callback, to conform with established jargon etc?

(nevermind, it is fine)